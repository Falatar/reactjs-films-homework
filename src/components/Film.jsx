import React from 'react';
import PropTypes from 'prop-types';
import style from './Film.scss';

function Film({
  name, genrePtime, rating,
}) {
  return (
    <div className={style.film}>
      <big>{name}</big>
      <h1>{genrePtime}</h1>
      <h1>{rating}</h1>
    </div>
  );
}

Film.defaultProps = {
  name: 'Can\'t find property "name"',
  genrePtime: ['unknown'],
  rating: 0,
};

Film.propTypes = {
  name: PropTypes.string,
  genrePtime: PropTypes.string,
  rating: PropTypes.number,
};

export default Film;
