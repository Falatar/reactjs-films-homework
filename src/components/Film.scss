div.film {
    display: flex;
    flex-direction: column;
    border: 0;
    padding: 0;
    margin: 0 0 1vh 6vw;
}

.film {
    big {
        width: 100%;
        font-size: 2em;
        color: white;
        font-weight: bold;
        margin: 0;
    }

    h1 {
        font-size: 1.2em;
        color: white;
        margin: 0 0 1vh 0;
    }
}