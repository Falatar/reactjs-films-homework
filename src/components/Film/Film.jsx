import React from 'react';
import PropTypes from 'prop-types';
import style from './Film.scss';
import Rate from '../Rate';

function Film({
  name, genrePtime, rating,
}) {
  return (
    <div className={style.Film}>
      <big className={style.Name}>{name}</big>
      <h1 className={style.Genres}>{genrePtime}</h1>
      <div className={style.RatePanel}>
        <Rate rating={rating} />
        <h1 className={style.Rate}>{rating.toFixed(1)}</h1>
      </div>
    </div>
  );
}

Film.defaultProps = {
  name: 'Can\'t find property "name"',
  genrePtime: ['unknown'],
  rating: 0,
};

Film.propTypes = {
  name: PropTypes.string,
  genrePtime: PropTypes.string,
  rating: PropTypes.number,
};

export default Film;
