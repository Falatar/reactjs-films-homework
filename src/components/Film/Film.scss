div.film {
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    display: flex;
    flex-direction: column;
    border: 0;
    padding: 0;
    margin: 0 0 1vh 6vw;
}

.film {
    big {
        width: 100%;
        font-size: 2em;
        color: white;
        font-weight: bold;
        margin: 0;
    }

    h1 {
        font-size: 1em;
        color: white;
        margin: 0 0 1vh 0;
        font-weight: lighter;
    }

    div.rate {
        display: flex;
        flex-direction: row;
        align-items: center;
        border: 0;
        padding: 0;
        margin: 0;
    }

    .rate {
        h1 {
            height: 1.1em;
            width: 2em;
            margin: 0 0 0 1em;
            padding: 0;
            border: 1px solid;
            border-color: deepskyblue;
            text-align: center;
        }
    }
}