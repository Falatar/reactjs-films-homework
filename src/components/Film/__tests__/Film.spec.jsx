import React from 'react';
import renderer from 'react-test-renderer';
import Film from '../Film';

it('renders correctly', () => {
  const tree = renderer
    .create(<Film name="nemo" genrePtime="empty" rating={0.0} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});