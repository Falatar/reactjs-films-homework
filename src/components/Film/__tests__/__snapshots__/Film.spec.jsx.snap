// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film renders correctly 1`] = `
<div
  className="film"
>
  <big
    className="name"
  >
    nemo
  </big>
  <h1
    className="genres"
  >
    empty
  </h1>
  <div
    className="rate__panel"
  >
    <div
      className="rate"
    >
      <span
        className="void_star"
      />
      <span
        className="void_star"
      />
      <span
        className="void_star"
      />
      <span
        className="void_star"
      />
      <span
        className="void_star"
      />
    </div>
    <h1
      className="rate"
    >
      0
    </h1>
  </div>
</div>
`;
